test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -132,66 +133,130 @@ var _ = Describe("Mixedcpus", Ordered, func() {
-			cmd := printMixedCPUsEnvCmd()
-			output, err := pods.ExecCommandOnPod(testclient.K8sClient, p, "", cmd)
-			Expect(err).ToNot(HaveOccurred(), "failed to execute command on pod; cmd=%q pod=%q", cmd, client.ObjectKeyFromObject(p).String())
-			isolatedAndShared := strings.Split(string(output), "\r\n")
-			isolated, err := cpuset.Parse(isolatedAndShared[0])
-			Expect(err).ToNot(HaveOccurred())
-			Expect(isolated.IsEmpty()).ToNot(BeTrue())
-			shared, err := cpuset.Parse(isolatedAndShared[1])
-			ppShared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
-			Expect(err).ToNot(HaveOccurred())
-			Expect(shared.Equals(ppShared)).To(BeTrue(), "OPENSHIFT_SHARED_CPUS value not equal to what configure in the performance profile."+
-				"OPENSHIFT_SHARED_CPUS=%s spec.cpu.shared=%s", shared.String(), ppShared.String())
+
+		When("Modifying the shared CPUs in the performance profile", func() {
+			BeforeEach(func() {
+
+				isolated := mustParse(string(*profile.Spec.CPU.Isolated))
+				shared := mustParse(string(*profile.Spec.CPU.Shared))
+				// select one arbitrary
+				isolatedCore := mustParse(strconv.Itoa(isolated.List()[0]))
+				sharedCore := mustParse(strconv.Itoa(shared.List()[0]))
+				// swap one from the isolated with one from the shared
+				//  to create new shared cpuset
+				newIsolated := isolated.Difference(*isolatedCore).Union(*sharedCore)
+				newShared := shared.Difference(*sharedCore).Union(*isolatedCore)
+				profile.Spec.CPU.Isolated = cpuSetToPerformanceCPUSet(&newIsolated)
+				profile.Spec.CPU.Shared = cpuSetToPerformanceCPUSet(&newShared)
+				By("applying new performanceProfile")
+				testprofiles.UpdateWithRetry(profile)
+				mcp, err := mcps.GetByProfile(profile)
+				Expect(err).ToNot(HaveOccurred())
+				By("waiting for mcp to catch up")
+				mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+				mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+				Expect(testclient.Client.Get(ctx, client.ObjectKeyFromObject(profile), profile))
+				testlog.Infof("new isolated CPU set=%q\nnew shared CPU set=%q", newIsolated.String(), newShared.String())
+				// we do not bother to revert the profile at the end of the test, since its irrelevant which of the cpus are shared
+			})
+
+			It("should contains the updated values under the container", func() {
+				rl := &corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("1"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+					sharedCpusResource:    resource.MustParse("1"),
+				}
+				p, err := createPod(ctx, testclient.Client, testutils.NamespaceTesting,
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err).ToNot(HaveOccurred())
+
+				By("checking cgroups under the container")
+				ppShared := mustParse(string(*profile.Spec.CPU.Shared))
```

Comment:
here we are comparing actual ppShared (because of [LOC 222](https://github.com/openshift/cluster-node-tuning-operator/pull/892/commits/78f8ae65fa117c188803f8dd0f2003251d5e676a#diff-a8e5e2d088216d6686138de966b7d90e5f2895bfa8e8b7fc6b9bcdd5b0397d69R222)) with actual pod values. This can hide (the test will pass) the bug of not having the PP shared cpus updated as expected which is the main goal of these set of tests.

Reply:
Not sure I follow, we are comparing the new shared cpus set from the profile with the test pod.
Could you please elaborate a little bit?

Reply:
ah I see this is addressed in line 219 (but with expect with no conditions, left a comment there)

Reply:
we are about to compare the updated value from actual PP instead of the expected. 
what will happen in a case updating the shared cpus wasn't actually reflected in the new profile? 
we need to compare the expected value with the actual one.

Reply:
Fixed. Thanks